addons/stock/models/stock_inventory.py

```diff
@@ -359,16 +366,46 @@ def _run_conflict_inventory_tasks(self):
             Note that an in progress inventory will prevent an update/unlink when its
             corresponding values are out of date.
         """
-        # negative quantity check
         existing_conflict_invs = self.search([('state', 'in', ['draft', 'confirm']),
                                               ('is_conflict_inventory', '=', True),
                                               ('company_id', 'in', self.env.user.company_ids.ids)])
-        updated_neg_invs = self.env['stock.quant'].search([('quantity', '<', 0.0),
-                                                                      ('location_id.usage', 'in', ['internal', 'transit']),
-                                                                      ('company_id', 'in', self.env.user.company_ids.ids)])\
-                                                             ._create_or_update_neg_inv(existing_conflict_invs)
+        # negative quantity check
+        updated_invs = self.env['stock.quant'].search([('quantity', '<', 0.0),
+                                                       ('location_id.usage', 'in', ['internal', 'transit']),
+                                                       ('company_id', 'in', self.env.user.company_ids.ids)])\
+                                              ._create_or_update_neg_inv(existing_conflict_invs)
+
+        # conflicting lots check
+        domain = [('location_id.usage', 'in', ['internal', 'transit']),
+                  ('lot_id', '!=', False),
+                  ('product_id.tracking', '=', 'serial'),
+                  ('quantity', '!=', 0.0),
+                  ('company_id', 'in', self.env.user.company_ids.ids)]
```

Comment:
maybe company_id should be a kwargs

Reply:
Switched it to a kwargs value of "company_domain" that is passed only when `_run_conflict_inventory_tasks` is called by the scheduler (since it varies depending on what triggers the scheduler) and default to `[('company_id', 'in', self.env.user.company_ids.ids)]` otherwise.